app/src/main/java/dev/sebastiano/bundel/preferences/DataStorePreferenceStorage.kt

```diff
@@ -0,0 +1,115 @@
+@file:Suppress("UnusedImports") // TODO bug in detekt 1.17.1 flags unused import incorrectly
+
+package dev.sebastiano.bundel.preferences
+
+import androidx.datastore.core.DataStore
+import dev.sebastiano.bundel.preferences.schedule.TimeRange
+import dev.sebastiano.bundel.preferences.schedule.TimeRangesSchedule
+import dev.sebastiano.bundel.preferences.schedule.WeekDay
+import dev.sebastiano.bundel.proto.BundelPreferences
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+import timber.log.Timber
+import java.io.IOException
+import java.time.LocalTime
+
+internal class DataStorePreferenceStorage(
+    private val dataStore: DataStore<BundelPreferences>
+) : PreferenceStorage {
+
+    override fun isCrashlyticsEnabled(): Flow<Boolean> =
+        dataStore.data.map { it.isCrashlyticsEnabled }
+            .catch { throwable ->
+                when (throwable) {
+                    is IOException -> {
+                        Timber.e(throwable)
+                        emit(true)
+                    }
+                    else -> {
+                        throw throwable
+                    }
+                }
+            }
+
+    override suspend fun setIsCrashlyticsEnabled(enabled: Boolean): Boolean =
```

Comment:
Checking the code, it looks like the return type is never used.
How about dropping it for all `set...` calls?